format/spec.md

```diff
@@ -193,10 +193,38 @@ Notes:
+
+Default values are encoded in JSON format. The representation depends on the type of the corresponding field. The mapping of types and their corresponding default value JSON representation is described in the following table.
+
+| type               | json type     | example                                | note                                                                                                                                                                                                                                         |
+|--------------------|---------------|----------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **`boolean`**      | **`boolean`** | true                                   |                                                                                                                                                                                                                                              |
+| **`int`**          | **`number`**  | 1                                      |                                                                                                                                                                                                                                              |
+| **`long`**         | **`number`**  | 1                                      |                                                                                                                                                                                                                                              |
+| **`float`**        | **`number`**  | 1.1                                    |                                                                                                                                                                                                                                              |
+| **`double`**       | **`number`**  | 1.1                                    |                                                                                                                                                                                                                                              |
+| **`decimal(P,S)`** | **`string`**  | "0x3162"                               | we use hexadecimal byte literals to encode bytes, with prefix `0x`, the byte array contain the two's-complement representation of the `unscaled` integer value in big-endian byte order, the actual value will be `unscaled * 10 ^ (-scale)` |
```

Comment:
In Iceberg, we avoid using personal pronouns ("we" or "I") in specs and docs. It is unclear who "we" is, and avoiding it ends up producing more direct documentation that is easily understood.

For example, here you should state that the values is "Stored as the two's-complement big-endian binary using the minimum number of bytes, converted to a hexadecimal string prefixed by 0x".

You should use wording mostly taken from Appendix D and adapted for JSON.

@RussellSpitzer, what do you think about base64 encoding or other representations? Do we care about the size here?

Reply:
For decimals I feel like we probably aren't saving that much space here, even with dozens of decimal defaults we would still probably be dominated by other entries in metadata.json and we have to string parse the whole thing before we get this value anyway. I think if we actually compressed metadata.json we probably would do better than smaller tuning like this if we were worried about space.

I think storing decimal values as a string is probably still ok and easier to read for humans.

Reply:
Sounds good to me! I was on the fence about this.

Do we need the `0x` prefix, or is it just nice to have that?